examples/service-catalog/service-catalog.yaml

```diff
@@ -0,0 +1,411 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: service-catalog
+objects:
+
+- kind: ServiceAccount
+  apiVersion: v1
+  metadata:
+    name: service-catalog-controller
+
+- kind: ClusterRoleBinding
+  apiVersion: v1
+  metadata:
+    name: template-broker-caller-binding
+  roleRef:
+    name: system:openshift:templateservicebroker-client
+  groupNames:
+  - system:unauthenticated
+  - system:authenticated
+  - system:anonymous
+
+- kind: ClusterRole
+  apiVersion: v1
+  metadata:
+    name: namespace-viewer
+  rules:
+  - apiGroups:
+    resources:
+    - namespaces
+    verbs:
+    - list
+    - watch
+    - get
+- kind: ClusterRoleBinding
+  apiVersion: v1
+  metadata:
+    name: service-catalog-namespace-viewer-binding
+  roleRef:
+    name: namespace-viewer
+  userNames:
+    - system:serviceaccount:service-catalog:default
+- kind: ClusterRoleBinding
+  apiVersion: v1
+  metadata:
+    name: service-catalog-controller-namespace-viewer-binding
+  roleRef:
+    name: namespace-viewer
+  userNames:
+    - system:serviceaccount:service-catalog:service-catalog-controller
+
+- kind: ClusterRole
+  apiVersion: v1
+  metadata:
+    name: secret-admin
```

Comment:
Also a rule can contain multiple resources and I think a cartesian product is used internally.  Thus 2 rules can be condensed when they are equal in all respects except the resources (and don't need to be separated for different bindings).  I think secret-admin and podpreset-admin are cases in point - they don't need to be separate rules on a clusterrole, but can be 2 resources in 1 rule in 1 clusterrole.